packages/grafana-ui/src/components/ColorPicker/NamedColorsPalette.tsx

```diff
@@ -25,15 +28,30 @@ export const NamedColorsPalette = ({ color, onChange, theme }: NamedColorsPalett
   });
 
   return (
-    <div
-      style={{
-        display: 'grid',
-        gridTemplateColumns: 'repeat(3, 1fr)',
-        gridRowGap: '24px',
-        gridColumnGap: '24px',
-      }}
-    >
-      {swatches}
-    </div>
+    <VerticalGroup spacing="md">
+      <div
+        style={{
+          display: 'grid',
+          gridTemplateColumns: 'repeat(3, 1fr)',
+          gridRowGap: theme.spacing(2),
+          gridColumnGap: theme.spacing(2),
+          flexGrow: 1,
+        }}
+      >
+        {swatches}
+        <ColorSwatch
+          isSelected={color === 'transparent'}
+          color={'rgba(0,0,0,0)'}
+          label="Transparent"
+          onClick={() => onChange('transparent')}
+        />
+        <ColorSwatch
+          isSelected={color === 'text'}
+          color={theme.colors.text.primary}
+          label="Text color"
+          onClick={() => onChange('text')}
+        />
```

Comment:
Don't get why the text swatch is here. 

Another thing - since the color picker is used in few places - I am not sure having the transparen/text swatches everywhere makes sens. WDYT about having those only for the threshold's color picker - since they make sense there only ATM. 

Alternatively they can serve as "hide but preserve series" but I don't think this a way we want it to be used. 

And when you set the series color to transparent, the legend icon is not visible, making it hard to change the color back. Let's enable it for thresholds only?

Reply:
> Don't get why the text swatch is here.

why not? it's a very useful color to pick, to get the text color independent of theme (think stat/gauge panel when you wan the value to have text color)

> Another thing - since the color picker is used in few places - I am not sure having the transparen/text swatches everywhere makes sens. WDYT about having those only for the threshold's color picker - since they make sense there only ATM. 

They are both very useful to stat panel as well

Reply:
Ok, I get that. But in this case I'm looking from the POV of Legend color picker in which I think neither transparent nor text make much sense. That's why I'm proposing these additional color options to be optional and intentional rather than default for every color picker.

Reply:
ah, maybe text color makes some sense but transparent not so much.

Reply:
Yea agree, let's hide these options from legend color picker